controllers/usersController.js

```diff
@@ -4,81 +4,61 @@ const User = db.users;
 const bcrypt = require("bcrypt");
 const saltRounds = 12;
 const { ERROR_MSG } = require("../utils/const");
+const addNewUser = require("../services/users.services");
+const redisClient = require("../middlewares/redis");
 
 function createNewUser(req, res, next) {
   const { email, password, role, firstname, lastname, gender, dob } = req.body;
 
-  bcrypt
-    .hash(password, saltRounds)
-    .then((hash) => {
-      let password = hash;
-      User.create({
-        email,
-        password,
-        role,
-        firstname,
-        lastname,
-        gender,
-        dob,
-      })
-        .then((data) => {
-          res.status(201).send({ data });
-        })
-        .catch((err) => {
-          logger.error(ERROR_MSG, err);
-          next(err);
-        });
+  addNewUser(email, password, role, firstname, lastname, gender, dob)
+    .then((user) => {
+      res.status(201).send({ user });
     })
     .catch((err) => {
-      logger.error("Unable to hash password", err);
+      logger.error(ERROR_MSG, err);
       next(err);
     });
 }
-function verifyUser(req, res, next) {
+async function verifyUser(req, res, next) {
   const { token } = req.params;
   const { email } = req.body;
+  const OTP = await redisClient.get("otp");
```

Comment:
This can't work as you can always have more than one user generating/verifying OTP same time